include/vcpkg/vcpkgpaths.h

```diff
@@ -21,6 +21,8 @@
 #include <vcpkg/base/optional.h>
 #include <vcpkg/base/path.h>
 
+#include <vcpkg/portfileprovider.h>
```

Comment:
The other `vcpkgpaths.h` things try very hard to only forward declare things. I did this because:

1. `portfileprovider.h` itself looks fairly lightweight.
2. This avoids needing to edit most users of `VcpkgPaths::overlay_ports`.